From 423a1faf26e4cbe278ae783c4d03a985ca9b53d4 Mon Sep 17 00:00:00 2001
From: Stefan Chulski <stefanc@marvell.com>
Date: Thu, 27 Apr 2017 10:22:10 +0300
Subject: [PATCH 0988/1345] net: mvpp2x: add number RSS tables parameter

commit  ced6e04c73b6a3636853912fcb946b5aa35e08c8 from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

This patch add number RSS tables parameter.
Parameter required for sysfs support.

Change-Id: I65cd75ee2bb4106996b48d625108d7241255f631
Signed-off-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38960
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Omri Itach <omrii@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h      |    1 +
 drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c |    4 +++-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
index c77b47e..f7fa840 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x.h
@@ -529,6 +529,7 @@ struct mv_pp2x {
 	/* RX flow hash indir'n table, in pp22, the table contains the
 	* CPU idx according to weight
 	*/
+	u8 num_rss_tables; /* created for sysfs usage */
 	u32 rx_indir_table[MVPP22_RSS_TBL_LINE_NUM];
 	u32 l4_chksum_jumbo_port;
 
diff --git a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
index 2d6248e..53031cc 100644
--- a/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
+++ b/drivers/net/ethernet/marvell/mvpp2x/mv_pp2x_main.c
@@ -5579,8 +5579,10 @@ static int mv_pp2x_probe(struct platform_device *pdev)
 	}
 
 	/* Init PP22 rxfhindir table evenly in probe */
-	if (priv->pp2_version == PPV22)
+	if (priv->pp2_version == PPV22) {
 		mv_pp22_init_rxfhindir(priv);
+		priv->num_rss_tables = mv_pp2x_queue_mode * mv_pp2x_num_cos_queues;
+	}
 
 	/* Initialize ports */
 	for_each_available_child_of_node(dn, port_node) {
-- 
1.7.9.5

